addons/mail/static/src/core/common/message_model.js

```diff
@@ -33,9 +33,6 @@ export class Message extends Record {
     attachment_ids = fields.Many("ir.attachment", { inverse: "message" });
     author_id = fields.One("res.partner");
     author_guest_id = fields.One("mail.guest");
-    get author() {
-        return this.author_id || this.author_guest_id;
-    }
```

Comment:
So, if I understand correctly, you removed a getter to essentially replace it by the value it returns everywhere it was used.
Isn't that counterproductive with what is a getter?

Reply:
@thjo-odoo Besides the intention is for JS code to handle JS objects "message", "partner", "member", etc. as the matching python records "mail.message", "res.partner", "discuss.channel.member", etc. Removal of this getter goes in the direction to reduce custom abstractions in JS code.

The value of change is not immediately made clear with this PR, this is a small change in the right direction. We made this PR with only this change because this is easier to review and merge things bits by bits.

Reply:
More carefully checking the PR, I'm thinking we might want to keep the `author` getter, but only when used to compare records. When accessing fields, I think we should always split it and/or make a dedicated getter for getting the field value.